lib/validator.js

```diff
@@ -0,0 +1,166 @@
+
+/**
+ * Is this a string?
+ * @param input
+ * @returns {boolean}
+ */
+const isString = input => {
+  return typeof input === 'string';
+};
+
+/**
+ * Is this a number?
+ * @param input
+ * @returns {boolean}
+ */
+const isNumber = input => {
+  return typeof input === 'number';
+};
+
+/**
+ * Is this an array?
+ * @param input
+ * @returns {boolean}
+ */
+const isArr = input => {
+  return Array.isArray(input) === true;
+};
+
+/**
+ * Is this an object?
+ * @param input
+ * @returns {boolean}
+ */
+const isObject = input => {
+  if(Array.isArray(input) === false) {
+    return typeof input === 'object';
+  }
+};
+
+/**
+ * Is this a function?
+ * @param input
+ * @returns {boolean}
+ */
+const isFunction = input => {
+  return typeof input === 'function';
+};
+
+/**
+ * Is this a boolean?
+ * @param input
+ * @returns {boolean}
+ */
+const isBoolean = input => {
+  return typeof input === 'boolean';
+};
+
+/**
+ * Is this an array of strings?
+ * @param {string[]} input 
+ * @returns {boolean}
+ */
+const isArrayOfStrings = (input) => {
+  for(let i = 0; i < input.length; i++) {
+    if(typeof input[i] === 'string') {
+      return input;
+    }
+  }
+};
+
+/**
+ * Is this an array of number?
+ * @param {number[]} input 
+ * @returns {boolean}
+ */
+const isArrayOfNumbers = (input) => {
```

Comment:
a shorter way to do these ones is: 
``const isArrayOfNumbers = (input) => {
  return input.every(isNumber);
};``